src/main/java/edu/harvard/iq/dataverse/api/Access.java

```diff
@@ -632,10 +629,34 @@ public Response postDownloadDatafiles(String fileIds, @QueryParam("gbrecs") bool
     @GET
     @Produces({"application/zip"})
     public Response downloadAllFromLatest(@PathParam("id") String datasetIdOrPersistentId, @QueryParam("gbrecs") boolean gbrecs, @QueryParam("key") String apiTokenParam, @Context UriInfo uriInfo, @Context HttpHeaders headers, @Context HttpServletResponse response) throws WebApplicationException {
+        logger.info("inside download-all api");
         try {
-            DataverseRequest req = createDataverseRequest(findUserOrDie());
-            final Dataset retrieved = execCommand(new GetDatasetCommand(req, findDatasetOrDie(datasetIdOrPersistentId)));
-            final DatasetVersion latest = execCommand(new GetLatestAccessibleDatasetVersionCommand(req, retrieved));
+            User user = findUserOrDie(); 
+            DataverseRequest req = createDataverseRequest(user);
+            final Dataset retrieved = findDatasetOrDie(datasetIdOrPersistentId);
+            if (!(user instanceof GuestUser)) {
+                // If it is a guest user, let's not even bother looking up if a draft exists.
```

Comment:
this comment feels a little weird here, since it's in the if that is not guest user and does check for draft

Reply:
hmm. does it feel weird? - it basically explains why that "if" statement is there - as to not bother looking up... etc. but sure, I can change it/or take it outside the "if".